homeassistant/components/tuya/__init__.py

```diff
@@ -68,6 +77,30 @@
 )
 
 
+def _update_discovery_interval(hass, interval):
+    tuya = hass.data[DOMAIN].get(TUYA_DATA)
+    if not tuya:
+        return
+
+    try:
+        tuya.discovery_interval = float(interval)
+        _LOGGER.info("Tuya discovery device poll interval set to %s seconds", interval)
+    except ValueError as ex:
+        _LOGGER.warning(ex)
+
+
+def _update_query_interval(hass, interval):
+    tuya = hass.data[DOMAIN].get(TUYA_DATA)
+    if not tuya:
+        return
+
+    try:
+        tuya.query_interval = float(interval)
+        _LOGGER.info("Tuya query device poll interval set to %s seconds", interval)
+    except ValueError as ex:
+        _LOGGER.warning(ex)
```

Comment:
We shouldn't need to validate that its a float as it should be validated before we accept or store the input

Reply:
Not validating but just converting, anyway I removed the conversion. Try - except is because attribute setter in library can raise ValueError.